pkg/controllers/jobset_controller.go

```diff
@@ -368,11 +368,6 @@ func (r *JobSetReconciler) createJobs(ctx context.Context, js *jobset.JobSet, ow
 
 	// If pod DNS hostnames are enabled, create a headless service for the JobSet
 	if dnsHostnamesEnabled(js) {
-
-		// Don't try to create a headless service without a subdomain
-		if js.Spec.Network.Subdomain == "" {
-			return fmt.Errorf("a subdomain should be set if dns hostnames are enabled")
-		}
```

Comment:
better to set `js.Spec.Network.Subdomain = js.Name` if empty in the top level function Reconcile so that we don't need to do this check everywhere.

The e2e tests are failing because you didn't change https://github.com/kubernetes-sigs/jobset/blob/45429d558afa45813fc19c86991b13a17470b694/pkg/controllers/jobset_controller.go#L616C3-L616C35

Reply:
Good catch, thanks! Done.